mautrix/client/api/user_data.py

```diff
@@ -81,7 +81,7 @@ async def set_displayname(self, displayname: str, check_current: bool = True) ->
             displayname: The new display name for the user.
             check_current: Whether or not to check if the displayname is already set.
         """
-        if check_current and await self.get_displayname(self.mxid) == displayname:
+        if check_current and await self.get_displayname(self.mxid) == (displayname or None):
```

Comment:
I think this will now break in the opposite case (send unnecessary requests if get_displayname returns an empty string)

Reply:
I was wondering that. I'll patch it to normalise `""` to `None` in all cases